pkg/datapath/linux/node.go

```diff
@@ -1574,7 +1578,7 @@ func (n *linuxNodeHandler) deleteIPsec(oldNode *nodeTypes.Node) {
 	scopedLog := log.WithField(logfields.NodeName, oldNode.Name)
 	scopedLog.Debugf("Removing IPsec configuration for node")
 
-	nodeID := n.getNodeIDForNode(oldNode)
+	nodeID := n.GetNodeIDForNode(oldNode)
 	if nodeID == 0 {
```

Comment:
```suggestion
	nodeID := n.getNodeIDForNode(oldNode)
	if nodeID == nil {
```
Then we can get rid of function `GetNodeIDForNode`.

Reply:
yep nice catch.